src/Zomp.SyncMethodGenerator/AsyncToSyncRewriter.cs

```diff
@@ -2088,7 +2088,14 @@ private bool RemoveDeclarator(VariableDeclaratorSyntax variable)
     private TypeSyntax ProcessSyntaxUsingSymbol(TypeSyntax typeSyntax)
     {
         var typeSymbol = semanticModel.GetTypeInfo(typeSyntax).Type;
-        return typeSymbol is null ? typeSyntax : ProcessSymbol(typeSymbol).WithTriviaFrom(typeSyntax);
+
+        return typeSymbol switch
+        {
+            null => typeSyntax,
+            { TypeKind: TypeKind.Enum } when typeSyntax is QualifiedNameSyntax { Right: IdentifierNameSyntax r } && (r.Identifier.ValueText != typeSymbol.Name || typeSymbol.ToString() != typeSyntax.WithoutTrivia().ToString())

```

Comment:
I've added a quick check with `r.Identifier.ValueText != typeSymbol.Name` which is almost always true, because `typeSymbol.ToString()` is allocating a string. The edge case is if the enum value and enum have the same name, e.g.

```cs
enum Test { Test }
```

This would fail with `r.Identifier.ValueText != typeSymbol.Name`. I've added a unit test for this.